python/ray/serialization.py

```diff
@@ -263,6 +273,22 @@ def object_id_deserializer(serialized_obj):
                     self._register_cloudpickle_serializer(
                         id_type, id_serializer, id_deserializer)
 
+    def get_and_clear_contained_object_ids(self):
+        if not self._thread_local:
+            self._thread_local = threading.local()
```

Comment:
Why not set this in `__init__`?

Reply:
Ah I think I misunderstood the semantics of `threading.local()`. updating...